<!-- Link -->
<element name="link" required="+">
  <description>A physical link with inertia, collision, and visual properties. A link must be a child of a model, and any number of links may exist in a model.</description>

  <attribute name="name" type="string" default="__default__" required="1">
    <description>A unique name for the link within the scope of the model.</description>
  </attribute>

  <element name="gravity" type="bool" default="true" required="0">
    <description>If true, the link is affected by gravity.</description>
  </element>

  <element name="self_collide" type="bool" default="false" required="0">
    <description>If true, the link can collide with other links in the model.</description>
  </element>

  <element name="kinematic" type="bool" default="false" required="0">
    <description>If true, the link is kinematic only</description>
  </element>

  <element name="pose" type="pose" default="0 0 0 0 0 0" required="0">
    <description>This is the pose of the link reference frame, relative to the model reference frame.</description>
  </element>

  <element name="must_be_base_link" type="bool" default="false" required="0">
    <description>If true, the link will have 6DOF and be a direct child of world.</description>
  </element>

  <element name="velocity_decay" required="1">
    <description>Exponential damping of the link's velocity.</description>
    <element name="linear" type="double" default="0.0" required="1">
      <description>Linear damping</description>
    </element>
    <element name="angular" type="double" default="0.0" required="1">
      <description>Angular damping</description>
    </element>
  </element> <!-- End velocity decay -->

  <include filename="inertial.sdf" required="0"/>
  <include filename="collision.sdf" required="*"/>
  <include filename="visual.sdf" required="*"/>
  <include filename="sensor.sdf" required="*"/>
  <include filename="projector.sdf" required="*"/>
  <include filename="audio_sink.sdf" required="*"/>
  <include filename="audio_source.sdf" required="*"/>

</element> <!-- End Link -->
